From f5f219759a3c01b79ec3c50d1a90c589b888c7fc Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Wed, 30 Jan 2019 15:37:28 -0800
Subject: [PATCH 0926/1051] soc: octeontx2: NPC KPU profile fix

This patch fixes a parser error when KPU parses incoming IPSec
ESP and AH packets

Change-Id: I2a19d6f090c4a70a9972b8cb5a92ec521cc7b3ab
Signed-off-by: Hao Zheng <haoz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3047
Reviewed-by: Hao Zheng <Hao.Zheng@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/npc_profile.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/npc_profile.h b/drivers/soc/marvell/octeontx2/npc_profile.h
index 0a27a071fd47..71ba06c15890 100644
--- a/drivers/soc/marvell/octeontx2/npc_profile.h
+++ b/drivers/soc/marvell/octeontx2/npc_profile.h
@@ -8618,7 +8618,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_ESP, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -8626,7 +8626,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_AH, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -8722,7 +8722,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_ESP, 0, 1,
 		NPC_LID_LC, NPC_LT_LC_IP_OPT,
 		0,
@@ -8730,7 +8730,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		0, 0, 0, 0, 0,
+		0, 0, 0, 2, 0,
 		NPC_S_KPU8_AH, 0, 1,
 		NPC_LID_LC, NPC_LT_LC_IP_OPT,
 		0,
-- 
2.17.1

